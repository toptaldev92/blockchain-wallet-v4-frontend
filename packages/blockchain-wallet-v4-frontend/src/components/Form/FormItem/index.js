import React from 'react'
import styled from 'styled-components'

const Wrapper = styled.div`
  width: ${props => props.width || '100%'};
  &:not(:first-child) {
    margin-top: 10px;
  }
`

const FormGroup = props => {
  const { children, ...rest } = props

  return (
    <Wrapper {...rest}>
      {children}
    </Wrapper>
  )
}

export default FormGroup
