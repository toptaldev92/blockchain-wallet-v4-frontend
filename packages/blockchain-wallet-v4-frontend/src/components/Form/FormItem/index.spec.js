import React from 'react'
import FormItem from './index.js'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
jest.mock('blockchain-info-components', () => ({ Text: 'text' }))

describe('FormItem', () => {
  it('renders correctly', () => {
    const props = { width: '10%' }
    const component = renderer.create(<FormItem {...props} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('width', '10%')
  })
})
