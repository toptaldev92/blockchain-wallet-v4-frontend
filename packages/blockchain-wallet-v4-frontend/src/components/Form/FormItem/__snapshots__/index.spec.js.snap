// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItem renders correctly 1`] = `
.c0 {
  position: relative;
  width: 10%;
}

<div
  className="c0"
  width="10%"
/>
`;
